internal/config/db.go

```diff
@@ -1,16 +1,24 @@
 package config
 
 import (
+	"fmt"
+
 	"github.com/b-open/jobbuzz/pkg/model"
 	"gorm.io/driver/mysql"
 	"gorm.io/gorm"
 )
 
 var Db gorm.DB
 
-func GetDb() (*gorm.DB, error) {
-	// TODO: 12-Factor the connection string
-	dsn := "jobbuzz:secret@tcp(127.0.0.1:3306)/jobbuzz?charset=utf8mb4&parseTime=True&loc=Local"
+func GetDb(dbConfig DbConfig) (*gorm.DB, error) {
+	dsn := fmt.Sprintf(
+		"%s:%s@tcp(%s:%s)/%s?charset=utf8mb4&parseTime=True&loc=Local",
+		dbConfig.Username,
+		dbConfig.Password,
+		dbConfig.Host,
+		dbConfig.Port,
+		dbConfig.Database,
+	)
```

Comment:
Seems like you can use config.FormatDSN instead

Code will look something like

dsnConfig := mysql.NewConfig()
dsnConfig.User = "user"
.....

mysql.Open(dsnConfig.FormatDSN(), ...)

Reply:
I don't get what you mean for this one